extensions/amazon-aws/src/glue.tsx

```diff
@@ -0,0 +1,291 @@
+          <List.Item.Detail.Metadata.Label title="Execution Time" text={glueJobRun.ExecutionTime?.toString() + "s"} />
+          <List.Item.Detail.Metadata.Label title="Started on" text={glueJobRun.StartedOn?.toUTCString()} />
+          <List.Item.Detail.Metadata.Label title="Completed On" text={glueJobRun.CompletedOn?.toUTCString() || ""} />
+          <List.Item.Detail.Metadata.Label
+            title="Error Message"
+            text={{ color: Color.Red, value: glueJobRun.ErrorMessage || "" }}
+          />
+          <List.Item.Detail.Metadata.Label title="Arguments" text={JSON.stringify(glueJobRun.Arguments)} />
+          <List.Item.Detail.Metadata.Label title="DPU Seconds" text={(glueJobRun.DPUSeconds?.toString() || "") + "s"} />
+          <List.Item.Detail.Metadata.Label title="Attempt" text={glueJobRun.Attempt?.toString()} />
+          <List.Item.Detail.Metadata.Separator />
+          <List.Item.Detail.Metadata.Label title="Worker Type" text={glueJobRun.WorkerType} />
+          <List.Item.Detail.Metadata.Label title="Number of Workers" text={glueJobRun.NumberOfWorkers?.toString()} />
+          <List.Item.Detail.Metadata.Label title="Timeout" text={glueJobRun.Timeout?.toString() + "s"} />
+          <List.Item.Detail.Metadata.Label title="Profile Name" text={glueJobRun.ProfileName || ""} />
+          <List.Item.Detail.Metadata.Separator />
+          <List.Item.Detail.Metadata.TagList title="Versions">
+            <List.Item.Detail.Metadata.TagList.Item text={"Glue V" + glueJobRun.GlueVersion} color={"#eed535"} />
+          </List.Item.Detail.Metadata.TagList>
+          <List.Item.Detail.Metadata.Label title="Log Group Name" text={glueJobRun.LogGroupName} />
+          <List.Item.Detail.Metadata.Label title="Id" text={glueJobRun.Id} />
+        </List.Item.Detail.Metadata>
+      }
+    />
+  );
+}
+
+async function RunGlueJob(glueJobName: string, mutate: MutatePromise<GlueJobRun[] | undefined>) {
+  mutate(new GlueClient({}).send(new StartJobRunCommand({ JobName: glueJobName })), {
+    optimisticUpdate(data) {
+      return data;
+    },
+  })
+    .then((response) => {
+      showToast({ style: Toast.Style.Success, title: "Triggered Glue job", message: "JobRunId: " + response.JobRunId });
+    })
+    .catch((error) => {
+      showFailureToast(error, { title: "Unable to trigger Glue Job" });
+    });
+}
+
+function GlueJobDefinition({ glueJobName: glueJobName }: { glueJobName: string }) {
+  const { data: glueJobDetails, isLoading } = useCachedPromise(fetchJobDetails, [glueJobName]);
+  const description = glueJobDetails?.Job?.Description
+    ? glueJobDetails?.Job?.Description!.toString()
+    : `No Descrtiption`;
+  return (
+    <Detail
+      navigationTitle="Glue Job Definition"
+      markdown={
+        `# ` +
+        glueJobName +
+        `\n\n` +
+        description +
+        `\n\n` +
+        `___\n## Job Definition\n` +
+        `\`\`\`json\n` +
+        JSON.stringify(glueJobDetails?.Job, null, 2) +
```

Comment:
Keep in mind if this definition gets large, there might be rendering delays for the `Detail` view, and in worst case scenarios might cause JS heap memory issues. I faced this problem when I was handling state machine definitions or cloud formation templates the same way (but not sure if glue job definitions will get that large). 

Can we check the response length and use a max threshold where it doesn't get slow. After a max threshold response length say 10000 characters we can just show on the Detail view that the definition was too long to display and prompt user to either copy it to clipboard or see it in console.

Reply:
I dont think that this i going to cause any issues. Glue definitions are, other than statemachines and cloud formation templates, not implementation specific. So they do not grow in size and actually its not an issue.